sycl/test-e2e/Basic/built-ins/marray_math.cpp

```diff
@@ -115,8 +157,35 @@ int main() {
   TEST3(sycl::nan, float, 3, ma7);
   if (deviceQueue.get_device().has(sycl::aspect::fp64))
     TEST3(sycl::nan, double, 3, ma8);
-  TEST(sycl::half_precision::exp10, float, 2, EXPECTED(float, 10, 100), 0.1,
-       ma1);
+
+  TEST4(sycl::half_precision::sin, sycl::half, 3,
+        EXPECTED(sycl::half, 0.98545f, -0.871576f, -0.832267f), ma6);
```

Comment:
According to the SYCL 2020 specification, the `sycl::half_precision` math functions all take `float` or either an marray, vec, or swizzled vec of `float`.

Additionally, the ULP expectation of all of these should be 8192 according to the spec.

Reply:
Yes